test-network-function/lifecycle/suite.go

```diff
@@ -300,6 +300,19 @@ func shutdownTest(podNamespace, podName string) {
 	test.RunAndValidate()
 }
 
+func testNodeDrain(env *config.TestEnvironment, nodeName string) {
+	// We need to delete all Oc sessions because the drain operation is often deleting oauth-openshift pod
+	// This results in lost connectivity for oc sessions
+	env.ResetOc()
+	// drain node
+	drainNode(nodeName)
+	// Ensure the node is uncordoned before exiting the function
+	defer uncordonNode(nodeName)
```

Comment:
i think we need to call the wait function again as part of the defer block

Reply:
Why should we? If waitFor..() succeds, we don't need to run it again, and if it fails, it will fail twice if we call it inside the defer block.

We have an extra waitFor...() call outside the for loop, in case there is some operator moving pods around when we uncordon the last node, shouldn't that be enough?

Reply:
no, the one outside should be removed as it's not reliable, it may be skipped if we panic. we do need to do it at least once in the loop so we know if the test result is success or failure and another time in the fail/error cases after uncordon. Putting it in defer would make it do the second one in all cases, but it shouldn't hurt in success case.

Reply:
Aha, I got it now. Thanks for the explanation! I'll update the PR to reflect this.